extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/ColorSplasher.pushbutton/script.py

```diff
@@ -164,133 +142,136 @@ def Execute(self, uiapp):
-                                view.RemoveFilter(filt_id)
-                                try:
-                                    new_doc.Delete(filt_id)
-                                except:
-                                    pass
-                except Exception as e:
-                    pass
-                #Reset visibility
-                for id in collector:
-                    view.SetElementOverrides(id, ogs)
-                t.Commit()
+                ogs = DB.OverrideGraphicSettings().Dispose()
+                ogs = DB.OverrideGraphicSettings()
+                collector = DB.FilteredElementCollector(new_doc, view.Id).WhereElementIsNotElementType().WhereElementIsViewIndependent().ToElementIds()
+                with revit.Transaction(new_doc, "Reset colors in elements"):
+                    try:
+                        # Get and ResetView Filters
+                        sel_cat = wndw._categories.SelectedItem['Value']
+                        sel_par = wndw._listBox1.SelectedItem['Value']
+                        filter_name = sel_cat._name + "/"
+                        filters = view.GetFilters()
+                        if len(filters) != 0:
+                            for filt_id in filters:
+                                filt_ele = new_doc.GetElement(filt_id)
+                                if filt_ele.Name.StartsWith(filter_name):
+                                    view.RemoveFilter(filt_id)
+                                    try:
+                                        new_doc.Delete(filt_id)
+                                    except:
+                                        pass
+                    except Exception as e:
+                        pass
+                    # Reset visibility
+                    for id in collector:
+                        view.SetElementOverrides(id, ogs)
         except Exception as e:
             pass
             
     def GetName(self):
         return "Reset colors in elements"
 
-class createLegend(IExternalEventHandler):
+
+class createLegend(UI.IExternalEventHandler):
     def __init__(self):
         pass
         
     def Execute(self, uiapp):
         try:
             new_doc = uiapp.ActiveUIDocument.Document
-            #Get legend view
-            collector = FilteredElementCollector(new_doc).OfClass(Autodesk.Revit.DB.View).ToElements()
-            legends=[]
+            rvt_ver = int(uiapp.Application.VersionNumber)
+            # Get legend view
+            collector = DB.FilteredElementCollector(new_doc).OfClass(DB.View).ToElements()
+            legends = []
             for vw in collector:
-                if vw.ViewType == ViewType.Legend:
+                if vw.ViewType == DB.ViewType.Legend:
                     legends.Add(vw)
                     break
-            if len(legends)>0:
-                #Duplicate existing legend
-                t = Transaction(new_doc, "Create Legend")
+            if len(legends) > 0:
```

Comment:
invert the if condition as suggested above, but this method has another big problem, it is really huge!
could this branch be extracted into a method/function, and maybe split into smaller ones?